src/store/transactions/index.ts

```diff
@@ -85,6 +89,11 @@ export type TransactionPayloadWithAttachment = TransactionPayload & {
 export const isUnauthorizedTransactionState = atom<boolean>(false);
 export const transactionBroadcastErrorState = atom<string | null>(null);
 
+// TODO: consider alternatives
+// Implicitly selecting a tx based on `pendingTransactionState`s value seems
+// like it could easily be error-prone. Say this value doesn't get reset when it should.
+// The effect could be calamitous. A user would be changing settings for a stale, cached
+// transaction they've long forgotten about.
```

Comment:
I agree here, we should make an issue to refactor this? Can we simplify how many different transaction states we have in the app?